bgw-net/bgw-net-server/src/main/kotlin/tools/aqua/bgw/net/server/service/oauth/SpringConfiguration.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 The BoardGameWork Authors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package tools.aqua.bgw.net.server.service.oauth
+
+import org.springframework.context.annotation.Configuration
+import org.springframework.security.config.annotation.web.builders.HttpSecurity
+import org.springframework.security.config.annotation.web.builders.WebSecurity
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.web.util.matcher.RequestMatcher
+
+private const val LOGOUT_SUCCESS_URL = "/"
+
+/**
+ * Configures spring security, doing the following:
+ * - Bypass security checks for static resources,
+ * - Restrict access to the application, allowing only logged in users,
+ * - Set up the login form
+ */
+@Configuration
+@EnableWebSecurity
+class SecurityConfiguration() : WebSecurityConfigurerAdapter() {
+  /** Require login to access internal pages and configure login form. */
```

Comment:
Docs does not match what gets configured.

Reply:
fixed